### PR TITLE
[m0] Add skip conditions for m0

### PR DESCRIPTION
**Pull request by yaqiangz**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Add skip conditions for m0 in custom mark file

#### How did you verify/test it?
Run test cases.

### CODE REVIEW

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 1]
```diff
@@ -551,6 +557,15 @@ platform_tests/api/test_watchdog.py:
     conditions:
       - "asic_type in ['barefoot'] and hwsku in ['newport', 'montara'] or ('sw_to3200k' in hwsku)"
 
+#######################################
+#####           broadcom          #####
+#######################################
+platform_tests/broadcom:
+  skip:
+    reason: "M0 topo does not support platform_tests/broadcom"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:565]: Please update this: use asic type

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 2]
```diff
@@ -589,6 +604,24 @@ platform_tests/fwutil/test_fwutil.py::test_fwutil_auto:
...
+      - "platform in ['armhf-nokia_ixs7215_52x-r0']"
+
+#######################################
+#####           mellanox          #####
+#######################################
+platform_tests/mellanox:
+  skip:
+    reason: "M0 topo does not support platform_tests/mellanox"
+    conditions:
+      - "topo_name in ['m0']"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:623]: Use asic type

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 3]
```diff
@@ -589,6 +604,24 @@ platform_tests/fwutil/test_fwutil.py::test_fwutil_auto:
...
     reason: "Command not yet merged into sonic-utilites"
 
+#######################################
+#####          link_flap          #####
+#######################################
+platform_tests/link_flap/test_cont_link_flap.py:
+  skip:
+    reason: "Skip test_cont_link_flap for Nokia-7215"
+    conditions:
+      - "platform in ['armhf-nokia_ixs7215_52x-r0']"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:614]: can you check if we can enable this test?

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 4]
```diff
@@ -629,17 +662,35 @@ platform_tests/test_reboot.py:
     conditions:
       - "hwsku in ['Celestica-E1031-T48S4']"
 
+platform_tests/test_reboot.py::test_fast_reboot:
+  skip:
+    reason: "Skip test_fast_reboot for Nokia-7215"
+    conditions:
+      - "platform in ['armhf-nokia_ixs7215_52x-r0']"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:669]: Skip for m0

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 5]
```diff
@@ -629,17 +662,35 @@ platform_tests/test_reboot.py:
...
-    reason: "Skip power off reboot test for Wistron"
+    reason: "Skip power off reboot test for Wistron/Nokia-7215"
+    conditions:
+      - "hwsku in ['Celestica-E1031-T48S4'] or 'sw_to3200k' in hwsku or platform in ['armhf-nokia_ixs7215_52x-r0']"
+
+platform_tests/test_reboot.py::test_soft_reboot:
+  skip:
+    reason: "Skip test_soft_reboot for Nokia-7215"
+    conditions:
+      - "platform in ['armhf-nokia_ixs7215_52x-r0']"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:681]: skip for m0

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 6]
```diff
@@ -629,17 +662,35 @@ platform_tests/test_reboot.py:
...
+    reason: "Skip test_fast_reboot for Nokia-7215"
+    conditions:
+      - "platform in ['armhf-nokia_ixs7215_52x-r0']"
+
 platform_tests/test_reboot.py::test_power_off_reboot:
   skip:
-    reason: "Skip power off reboot test for Wistron"
+    reason: "Skip power off reboot test for Wistron/Nokia-7215"
+    conditions:
+      - "hwsku in ['Celestica-E1031-T48S4'] or 'sw_to3200k' in hwsku or platform in ['armhf-nokia_ixs7215_52x-r0']"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:675]: can you check what is the test intention? why we want to skip it

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 7]
```diff
@@ -629,17 +662,35 @@ platform_tests/test_reboot.py:
...
+    reason: "Skip test_soft_reboot for Nokia-7215"
+    conditions:
+      - "platform in ['armhf-nokia_ixs7215_52x-r0']"
+
+platform_tests/test_reboot.py::test_warm_reboot:
+  skip:
+    reason: "Skip test_warm_reboot for Nokia-7215"
     conditions:
-      - "hwsku in ['Celestica-E1031-T48S4'] or 'sw_to3200k' in hwsku"
+      - "platform in ['armhf-nokia_ixs7215_52x-r0']"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:687]: skip for m0

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 8]
```diff
@@ -629,17 +662,35 @@ platform_tests/test_reboot.py:
...
 
 platform_tests/test_reboot.py::test_watchdog_reboot:
   skip:
-    reason: "Skip watchdog reboot test for Wistron"
+    reason: "Skip watchdog reboot test for Wistron/Nokia-7215"
     conditions:
-      - "hwsku in ['Celestica-E1031-T48S4'] or 'sw_to3200k' in hwsku"
+      - "hwsku in ['Celestica-E1031-T48S4'] or 'sw_to3200k' in hwsku or platform in ['armhf-nokia_ixs7215_52x-r0']"
```
Comment (Blueve) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:693]: skip for m0, but we may need this enabled for m0 in future

**Review by Blueve — CHANGES_REQUESTED** [review]

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 6 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:666]: In 7215, we can't get reboot cause

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 1 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml]: Fixed

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 2 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml]: Fixed

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 4 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml]: Fixed

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 5 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml]: Fixed

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 7 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml]: Fixed

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 8 above)
Comment (yaqiangz) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml]: Fixed

**Review by Blueve — APPROVED** [review]

**Pull request merged**